packages/gestalt/src/Touchable.js

```diff
@@ -73,126 +85,187 @@ const getRoundingStyle = (r: Rounding): Style => {
   return identity();
 };
 
-export default class Touchable extends React.Component<Props> {
-  static propTypes = {
-    accessibilityControls: PropTypes.string,
-    accessibilityExpanded: PropTypes.bool,
-    accessibilityHaspopup: PropTypes.bool,
-    accessibilityLabel: PropTypes.string,
-    children: PropTypes.node,
-    disabled: PropTypes.bool,
-    fullHeight: PropTypes.bool,
-    fullWidth: PropTypes.bool,
-    mouseCursor: PropTypes.oneOf([
-      'copy',
-      'grab',
-      'grabbing',
-      'move',
-      'noDrop',
-      'pointer',
-      'zoomIn',
-      'zoomOut',
-    ]),
-    onBlur: PropTypes.func,
-    onFocus: PropTypes.func,
-    onTouch: PropTypes.func,
-    onMouseEnter: PropTypes.func,
-    onMouseLeave: PropTypes.func,
-    rounding: RoundingPropType,
-  };
-
-  handleKeyPress = (event: SyntheticKeyboardEvent<HTMLDivElement>) => {
-    const { disabled, onTouch } = this.props;
-    // Check to see if space or enter were pressed
-    if (
-      !disabled &&
-      onTouch &&
-      (event.charCode === SPACE_CHAR_CODE || event.charCode === ENTER_CHAR_CODE)
-    ) {
-      // Prevent the default action to stop scrolling when space is pressed
-      event.preventDefault();
-      onTouch({ event });
-    }
-  };
+function Touchable({
+  accessibilityControls,
+  accessibilityExpanded,
+  accessibilityHaspopup,
+  accessibilityLabel,
+  children,
+  disabled = false,
+  forwardedRef,
+  fullHeight,
+  fullWidth = true,
+  inline = false,
+  mouseCursor = 'pointer',
+  onBlur,
+  onFocus,
+  onMouseEnter,
+  onMouseLeave,
+  onTouch,
+  pressBackgroundEnabled = false,
+  pressScaleEnabled = false,
+  pressShadowEnabled = false,
+  rounding = 0,
+}: Props) {
+  const [isFocused, setFocused] = React.useState<boolean>(false);
+  const [isPressed, setPressed] = React.useState<boolean>(false);
+  const [coordinate, setCoordinate] = React.useState<Coordinate>({
+    x: 0,
+    y: 0,
+  });
+  const ignoreNextFocusEventRef = React.useRef<boolean>(false);
 
-  handleBlur = (event: SyntheticFocusEvent<HTMLDivElement>) => {
-    const { disabled, onBlur } = this.props;
-    if (!disabled && onBlur) {
-      onBlur({ event });
+  const classes = classnames(
+    styles.touchable,
+    toProps(getRoundingStyle(rounding)).className,
+    {
+      [styles.fullHeight]: fullHeight,
+      [styles.fullWidth]: !inline && fullWidth,
+      [styles.inline]: inline,
+      [styles[mouseCursor]]: !disabled,
+      [styles.pressScale]: !disabled && pressScaleEnabled && isPressed,
+      [styles.pressBackground]:
+        !disabled && pressBackgroundEnabled && (isPressed || isFocused),
+      [styles.pressShadow]:
+        !disabled && pressShadowEnabled && (isPressed || isFocused),
     }
-  };
+  );
 
-  handleFocus = (event: SyntheticFocusEvent<HTMLDivElement>) => {
-    const { disabled, onFocus } = this.props;
-    if (!disabled && onFocus) {
-      onFocus({ event });
-    }
-  };
+  return (
+    <div
```

Comment:
one difference here is that `TapArea` in pinboard supports an `href` prop, which will change it to an anchor.  options here for supporting that could be:
- update gestalt's `Link` to support touch
- update this to support an href prop (feels gross)

@christianvuerings @jackhsu978 wdyt?

Reply:
I thought about supporting `href` prop, but if we do that, we are going to a lot more props like `nofollow` and `target`. It's gonna be gross.

I think we should make components single responsibility. Touchable should just support the look & feel of a touchable UI element, but it should not care what's going to happen after touch.

For the link use case, this is what I have in mind (I tested in Pinboard and it seems to work fine):

```
<Touchable ...>
  <a href={href} rel={nofollow ? 'nofollow' : ''}>
    {children}
  </a>
</Touchable>
```

PS: Note that I am not using Gestalt's `<Link>` in the code snippet above because: (1) `Link` always underlines text on hover and (2) it doesn't support `nofollow` prop. We can make Gestalt's `<Link>` component support those, but that's a separate discussion.

Reply:
^yeah the problem with that (and why we originally added this to `TapArea`) is accessibility (having a button that wraps a link)

we could address this by adding touch support to `Link` - would definitely make the intent clearer (imo)

Reply:
`Touchable` uses a div element with role "button". What if we add a prop "role" which can be "button" or null, and make it "button" by default? (also, `tab-index` attribute would be 0 if role is `null`)

Would that work?

Reply:
Will address it with a new component called `TapLink`